test/active_record_doctor/detectors/missing_unique_indexes_test.rb

```diff
@@ -55,6 +55,25 @@ def test_present_unique_index
     refute_problems
   end
 
+  def test_missing_unique_index_and_sti
+    create_table(:users) do |t|
+      t.string :type
+      t.string :email
+      t.string :name
+    end.define_model do
+      validates :email, uniqueness: true
+    end
+
+    define_model(:Client, TransientRecord::Models::User) do
+      validates :name, uniqueness: true
+    end
+
+    assert_problems(<<~OUTPUT)
+      add a unique index on users(email) - validating uniqueness in the model without an index can lead to duplicates
+      add a unique index on users(name) - validating uniqueness in the model without an index can lead to duplicates
```

Comment:
After taking STI into account, these error messages became confusing. It's not clear which model is defining the validator. I suggest including the model name in the message, e.g.:

```
add a unique index on users(email) - validating uniqueness in User without an index can lead to duplicates
```